engine/collection/ingest/engine.go

```diff
@@ -23,13 +27,16 @@ import (
 // transactions are delegated to the correct collection cluster, and prepared
 // to be included in a collection.
 type Engine struct {
-	unit                 *engine.Unit
+	lm                   *lifecycle.LifecycleManager
```

Comment:
I think you could consider looking into using `component.ComponentManager` instead? It's pretty well documented, and if you grep the codebase there are already many examples of it in use.

It's a better fit for the `Component` interface, because the idea is supposed to be that cancelling the context passed into `Start` should stop the component. You shouldn't need to explicitly call `Done()` to stop it.

Reply:
I would maybe look at `p2p/network.go` as a specific example, because it demonstrates how you would implement public methods (e.g `Register` in the case of the network, but for engines an example of a public method would be `Process`) in a way that takes the context into account.

So that once the component has been shutdown by cancelling the context, calls made to public methods will return an error.

Reply:
👍  Thanks, I was working off the compliance engine as an [example](https://github.com/onflow/flow-go/blob/37781b8649ff483942163bbca14b17140c8997f4/engine/collection/compliance/engine.go#L36), `Network` looks like a better example.